### PR TITLE
use retryablehttp in DiscoverKeys and pkg/apk

### PR DESCRIPTION
**Pull request by imjasonh**
This should help us paper over some flakes we sometimes see with these requests.

### CODE REVIEW

File: pkg/apk/apk/implementation.go [hunk 1]
```diff
@@ -850,7 +850,10 @@ func DiscoverKeys(ctx context.Context, client *http.Client, auth auth.Authentica
 	if err := auth.AddAuth(ctx, discoveryRequest); err != nil {
 		return nil, err
 	}
-	discoveryResponse, err := client.Do(discoveryRequest)
+
+	rc := retryablehttp.NewClient()
+	rc.HTTPClient = client
+	discoveryResponse, err := rc.StandardClient().Do(discoveryRequest)
```
Comment (mattmoor) [inline pkg/apk/apk/implementation.go:856]: Given the way we wire in APK caching, I am terrified of changes here.

cc @jonjohnsonjr 👀

Comment (imjasonh) [reply pkg/apk/apk/implementation.go:856]: I agree!

Comment (jonjohnsonjr) [reply pkg/apk/apk/implementation.go:856]: same but I don't have any specific objections just bad vibes

**Review by jonjohnsonjr — APPROVED** [review]

**Comment by mepi262** [comment]
@jonjohnsonjr 
Would you tell me why this pull request is not merged?

**Comment by jonjohnsonjr** [comment]
> Would you tell me why this pull request is not merged?

Sorry, neglected to click the button.

**Comment by mepi262** [comment]
@jonjohnsonjr 
Thank you for your reply and action!

**Pull request merged**